src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -99,6 +107,67 @@ public IEnumerable<IOption> GetRegisteredOptions()
             return _lazyAllOptions.Value;
         }
 
+        public bool TryMapEditorConfigKeyToOption(string key, string? language, [NotNullWhen(true)] out IEditorConfigStorageLocation2? storageLocation, out OptionKey optionKey)
+        {
+            ImmutableDictionary<string, (IOption? option, IEditorConfigStorageLocation2? storageLocation)> temporaryOptions = s_emptyEditorConfigKeysToOptions;
+            ref var editorConfigToOptionsStorage = ref temporaryOptions;
```

Comment:
this scares me :)  any way to avoid all these ref-shenanigans?

Reply:
It may go away in the future, but overall I was quite happy with the way this turned out in the end.